Common/Data/Consolidators/RangeConsolidator.cs

```diff
@@ -0,0 +1,223 @@
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Python.Runtime;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using System;
+
+namespace QuantConnect.Data.Consolidators
+{
+    /// <summary>
+    /// This consolidator can transform a stream of <see cref="IBaseData"/> instances into a stream of <see cref="RangeBar"/>
+    /// </summary>
+    public class RangeConsolidator : BaseTimelessConsolidator
+    {
+        private bool _firstTick;
+        private decimal _minimumPriceVariation;
+        protected RangeBar CurrentRangeBar;
+
+        /// <summary>
+        /// Bar being created
+        /// </summary>
+        protected override TradeBar CurrentBar
+        {
+            get
+            {
+                return CurrentRangeBar;
+            }
+            set
+            {
+                CurrentRangeBar = (RangeBar)value;
+            }
+        }
+
+        /// <summary>
+        /// Range for each RangeBar, this is, the difference between the High and Low for each
+        /// RangeBar
+        /// </summary>
+        public decimal RangeSize { get; private set; }
+
+        /// <summary>
+        /// Number of MinimumPriceVariation units
+        /// </summary>
+        public decimal Range { get; private set; }
+
+        /// <summary>
+        /// Gets <see cref="RangeBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
+        /// </summary>
+        public override Type OutputType => typeof(RangeBar);
+
+        /// <summary>
+        /// Gets a clone of the data being currently consolidated
+        /// </summary>
+        public override IBaseData WorkingData => CurrentRangeBar?.Clone();
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public new event EventHandler<RangeBar> DataConsolidated;
+
+        /// <summary>
+        ///Initializes a new instance of the <see cref="RangeConsolidator" /> class.
+        /// </summary>
+        /// <param name="range">The constant value size of each bar, where the unit is the minimum price variation of the RangeBar's symbol</param>
+        public RangeConsolidator(decimal range)
+            : this(range, x => x.Value)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RangeConsolidator" /> class.
+        /// </summary>
+        /// <param name="range">The constant value size of each bar, where the unit is the minimum price variation of the RangeBar's symbol</param>
```

Comment:
Clarify the purpose of the range parameter: the number of minimum price variations a bar consists of.